launcher/src/flattened-devfile.ts

```diff
@@ -13,7 +13,9 @@ import { env } from "process";
 import * as jsYaml from "js-yaml";
 
 export interface Devfile {
-  projects: Project[];
+  projects?: Project[];
+  dependentProjects?: Project[];
+  starterProjects?: Project[];
```

Comment:
We likely need to wrap starterProjects in a getter:

1. Check for attribute `controller.devfile.io/use-starter-project: <name>` in flattened DevWorkspace
2. Return the only starterProject with `<name>` in attribute, if it exists

If the `controller.devfile.io/use-starter-project` attribute is not present in flattened spec, no starter projects will be cloned and so the workspace will contain nonexistent directories. This can be tested with e.g. the Go devfile in registry.devfile.io: https://registry.devfile.io/devfiles/go/1.0.2